zulipterminal/ui_tools/boxes.py

```diff
@@ -394,6 +409,7 @@ def keypress(self, size: urwid_Size, key: str) -> Optional[str]:
                 if self.to_write_box is None:
                     if header.focus_col == 0:
                         stream_name = header[0].edit_text
+                        self.set_stream_write_box_caption(stream_name)
```

Comment:
This is extra compared to the last version; which condition requires this?

Reply:
Yes. This is the condition when the user cycles the compose focus, and thus the colors need to be handled for that.
Also, in my updated current commit shortly, I'll update the colors also when the user presses command to **send** the message :)